polkadot/xcm/xcm-executor/src/lib.rs

```diff
@@ -624,6 +624,19 @@ impl<Config: config::Config> XcmExecutor<Config> {
 				Ok(())
 			},
 			DepositReserveAsset { assets, dest, xcm } => {
+				// we need to do this take/put cycle to solve wildcards and get exact assets to be
+				// weighed
+				let to_weigh = self.holding.saturating_take(assets.clone());
+				self.holding.subsume_assets(to_weigh.clone());
+
+				let mut message_to_weigh =
+					vec![ReserveAssetDeposited(to_weigh.into()), ClearOrigin];
+				message_to_weigh.extend(xcm.0.clone().into_iter());
+				let (_, fee) = validate_send::<Config::XcmSender>(dest, Xcm(message_to_weigh))?;
+				// set aside fee to be charged by XcmSender
+				let parked_fee = self.holding.saturating_take(fee.into());
```

Comment:
nit:

this can be extracted in some `fn extract_transport_fee()` or similar and reused